Pester Task: Trying to download in a isolated env. during run

### Azure DevOps Extension you are using
Pester Tasks

### Where are you running it?
On premise - Azure DevOps Server 

### Version of Extension/Task
Latest

### Expected behaviour and actual behaviour:
When starting a test run I get the following error messages:

**Unable to download from URI 'https://go.microsoft.com/fwlink/?LinkID=627338&clcid=0x409' to ''.
Unable to download the list of available providers. Check your internet connection.
System.Exception: No match was found for the specified search criteria for the provider 'Nuget'. The package provider requires 'PackageManagement' and 'Provider' tags. Please check if the specified package has the tags.**

This is an environment that has no internet access. 

### Steps to reproduce the problem

- Kick off the test
- script makes it to: **Adding additional module path [] to $env:PSModulePath**
- The list of errors generated are:

**##[warning]MSG:UnableToDownload «https://go.microsoft.com/fwlink/?LinkID=627338&clcid=0x409» «»
 ##[warning]Unable to download from URI 'https://go.microsoft.com/fwlink/?LinkID=627338&clcid=0x409' to ''.
 ##[warning]Unable to download the list of available providers. Check your internet connection.
 ##[warning]Unable to download from URI 'https://go.microsoft.com/fwlink/?LinkID=627338&clcid=0x409' to ''.
 ##[warning]Unable to download the list of available providers. Check your internet connection.
 ##[error]System.Exception: No match was found for the specified search criteria for the provider 'Nuget'. The package provider requires 'PackageManagement' and 'Provider' tags. Please check if the specified package has the tags.
 ##[error]System.Management.Automation.MethodInvocationException: Exception calling "ShouldContinue" with "2" argument(s): "The method or operation is not implemented." ---> System.NotImplementedException: The method or operation is not implemented.**

--- a/Extensions/Pester/readme.md
+++ b/Extensions/Pester/readme.md
@@ -52,3 +52,4 @@ Releases
 - 8.8.x - Add ScriptBlock parameter to allow running a script before running the tests. [Fixes #377](https://github.com/rfennell/vNextBuild/issues/377)
 - 8.11.x - Fixing Find-Module issues raised in [#412](https://github.com/rfennell/AzurePipelines/issues/412) and [415](https://github.com/rfennell/AzurePipelines/issues/415) by adding a check for AllowPrerelease switch on the cmdlet. If it's not available we'll fall back to the version of Pester that is shipped with the extension and write a warning that a newer version can't be used without a newer version of PowerShellGet being available.
 - 8.12.x - Updating built in version of Pester to 4.6.0. Remove check for AllowPrerelease as it's not needed. Fixes [#421](https://github.com/rfennell/AzurePipelines/issues/421)
+- 8.13.x - Update to better support offline build machines with no nuget installed. Falls back to shipped version of Pester. Fixes [#447](https://github.com/rfennell/AzurePipelines/issues/447)

--- a/Extensions/Pester/task/Pester.ps1
+++ b/Extensions/Pester/task/Pester.ps1
@@ -75,7 +75,13 @@ if ((Get-Module -Name PowerShellGet -ListAvailable) -and
         $null = Get-PackageProvider -Name NuGet -ErrorAction Stop
     }
     catch {
-        Install-PackageProvider -Name Nuget -RequiredVersion 2.8.5.201 -Scope CurrentUser -Force -Confirm:$false
+        try {
+            Install-PackageProvider -Name Nuget -RequiredVersion 2.8.5.201 -Scope CurrentUser -Force -Confirm:$false -ErrorAction Stop
+        }
+        catch {
+            Write-Host "##vos[task.logissue type=warning]Falling back to version of Pester shipped with extension. To use a newer version please update the version of PowerShellGet available on this machine."
+            Import-Module "$PSScriptRoot\4.6.0\Pester.psd1" -force
+        }
     }
     $NewestPester = Find-Module -Name Pester | Sort-Object Version -Descending | Select-Object -First 1
     If ((Get-Module Pester -ListAvailable | Sort-Object Version -Descending| Select-Object -First 1).Version -lt $NewestPester.Version) {

